src/smartcontracts/nativeSerializer.spec.ts

```diff
@@ -610,4 +610,59 @@ describe("test native serializer", () => {
             variadic: false,
         });
     });
+
+    it.only("should accept a mixed of values for boolen type", async () => {
+        const endpoint = AbiRegistry.create({
+            endpoints: [
+                {
+                    name: "foo",
+                    inputs: [
+                        {
+                            type: "bool",
+                        },
+                        {
+                            type: "bool",
+                        },
+                        {
+                            type: "bool",
+                        },
+                        {
+                            type: "bool",
+                        },
+                        {
+                            type: "bool",
+                        },
+                        {
+                            type: "bool",
+                        },
+                        {
+                            type: "bool",
+                        },
+                        {
+                            type: "bool",
+                        },
+                        {
+                            type: "bool",
+                        },
+                    ],
+                    outputs: [],
+                },
+            ],
+        }).getEndpoint("foo");
+
+        let typedValues = NativeSerializer.nativeToTypedValues(
+            [true, "true", "TRUE", 1, false, "false", "falseString", 0, 5],
```

Comment:
`"falseString"` can be `"foobar"` - test is thus more explicit about arbitrary strings.